mlflow/utils/autologging_utils/versioning.py

```diff
@@ -21,7 +21,7 @@
 
 
 def _check_version_in_range(ver, min_ver, max_ver):
-    return LooseVersion(min_ver) <= LooseVersion(ver) <= LooseVersion(max_ver)
+    return Version(min_ver) <= Version(ver) <= Version(max_ver)
```

Comment:
@harupy 

Should we directly skip dev or pre-release version ? 
i.e., skip `Version(ver).is_devrelease or Version(ver).is_prerelease`

Reply:
I don't think we should include them since we don't test them in the cross-version-tests

Reply:
Can you add tests to make sure `_check_version_in_range` works correctly for dev, pre, and post releases?